proto/qgb/query.proto

```diff
@@ -36,22 +37,26 @@ service Query {
 
   // DataCommitmentConfirm Queries the data commitment confirm by the commitment
   // and the validator address
+  // Returns nil if not found.
   rpc DataCommitmentConfirm(QueryDataCommitmentConfirmRequest)
       returns (QueryDataCommitmentConfirmResponse) {
     option (google.api.http).get =
         "/qgb/data_commitment/confirm/begin/"
         "{begin_block}/end/{end_block}/address/{address}";
   }
   // DataCommitmentConfirms Queries data commitment confirms by the actual
-  // commitment
+  // commitment.
+  // Too heavy, shouldn't mainly be used.
+  // Returns empty array if no element is found.
```

Comment:
yes, definitely, it has to do with indexing. If we start double indexing data using multiple keys, we can define iterators with more specific store ranges, and it would be way faster

Reply:
We should investigate however the storage cost if it is worth it compared to the gained performance

Reply:
opened issue: https://github.com/celestiaorg/celestia-app/issues/552